{{cpp/title|runtime_error}}
{{cpp/error/navbar}}
{{ddcl|header=stdexcept|
class runtime_error;
}}

定义作为异常抛出的对象类型。它报告源于程序范围之外，且无法轻易预测到的错误。

{{inheritance diagram/std-runtime_error}}

下列标准库组件会抛出 {{tt|std::runtime_error}} 类型异常：
* {{lc|std::locale::locale}}
* {{lc|std::locale::combine}}
{{rrev|since=c++20|
* {{c/core|std::chrono::current_zone}}
* {{c/core|std::chrono::get_tzdb_list}}
* {{c/core|std::chrono::get_tzdb}}
* {{c/core|std::chrono::locate_zone}}
* {{c/core|std::chrono::reload_tzdb}}
* {{c/core|std::chrono::remote_version}}
* {{c/core|std::chrono::tzdb::locate_zone}}
}}

另外，以下标准异常类型派生自 {{tt|std::runtime_error}}：
* {{lc|std::range_error}}
* {{lc|std::overflow_error}}
* {{lc|std::underflow_error}}
{{rrev|since=c++11|
* {{lc|std::ios_base::failure}}
* {{lc|std::regex_error}}
* {{lc|std::system_error}}
}}
{{rrev|since=c++17|
* {{lc|std::filesystem::filesystem_error}}
}}
{{rrev|since=c++20|
* {{c/core|std::chrono::ambiguous_local_time}}
* {{c/core|std::chrono::nonexistent_local_time}}
* {{c/core|std::format_error}}
}}

===成员函数===
{{cpp/error/exception/member of derived with message|runtime_error}}

{{cpp/error/exception/inherit}}

===缺陷报告===
{{dr list begin}}
{{dr list item|wg=lwg|dr=254|std=C++98|before=缺失了接受 {{c|const char*}} 的构造函数|after=已补充}}
{{dr list item|wg=lwg|dr=471|std=C++98|before={{tt|std::runtime_error}} 的复制的解释字符串由实现定义|after=它们与原 {{tt|std::runtime_error}} 对象的解释字符串相同}}
{{dr list end}}

{{langlinks|de|en|es|fr|it|ja|pt|ru}}